File: 2026-08/src/day_22/prms_42897/Solution2.py
def solution(money):
    n = len(money)

    def solve(l, r):
        prv2, prv1 = 0, 0

        for i in range(l, r):
            prv2, prv1 = prv1, max(prv1, prv2 + money[i])

        return prv1

    case1 = solve(0, n - 1)
    case2 = solve(1, n)
    return max(case1, case2)

File: 2026-08/src/day_22/prms_42897/Solution2.java
package day_22.prms_42897;

class Solution2 {
    public int solution(int[] money) {
        int n = money.length;

        int case1 = solve(money, 0, n - 1);
        int case2 = solve(money, 1, n);
        return Math.max(case1, case2);
    }

    static int solve(int[] money, int l, int r) {
        int prv2 = 0, prv1 = 0;

        for (int i = l; i < r; i++) {
            int cur = Math.max(prv1, prv2 + money[i]);
            prv2 = prv1;
            prv1 = cur;
        }

        return prv1;
    }
}
